src/pages/index.js

```diff
@@ -63,48 +100,158 @@ Array.from(document.forms).forEach((formElement) => {
+  popupCardAdd.open();
+});
+
 const popupCardAdd = new PopupWithForm(
   modalWindowCardAdd,
-  (event, cardInputs) => {
+  (event, { title, link }) => {
     event.preventDefault();
-    const cardElement = createCard(cardInputs);
-    cardSection.addItem(cardElement, "prepend"); // can still be used to add later cards, not just init ones
-    popupCardAdd.close();
-    formCardAdd.reset();
-    formValidators["card-add_form"].disableButton();
+    popupCardAdd.setRender(true);
+    api
+      .addNewCard(title, link)
+      .then((newCard) => {
+        const cardElement = createCard(newCard);
+        cardSection.addItem(cardElement, "prepend"); // can still be used to add later cards, not just init ones
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupCardAdd.close();
+        popupCardAdd.setRender(false);
+        formCardAdd.reset();
+        formValidators["card-add_form"].disableButton();
+      });
   },
-  configuration.inputSelector
+  configuration.inputSelector,
+  submitCardAdd,
+  "Creating..."
 );
 popupCardAdd.setEventListeners();
+/* END CARD ADD SECTION */
 
-buttonAddCard.addEventListener("click", () => {
-  // don't show errors upon opening the modal when no input is issued
-  popupCardAdd.open();
+/* CARD VIEW IMAGE SECTION */
+function handleCardPopupImage(card) {
+  // suggested by Sprint 7 project to keep this func in index.js for now
+  popupImage.open(card); // since "this" is prohibited outside of classes, and passing an arg to the func as a param to this func causes a lexical ref err, then in the class's handler func it would need to pass the ready-made getInfo() method to pass through unimpeded
+}
+const popupImage = new PopupWithImage(modalWindowCardView, {
+  image: modalImage,
+  caption: modalCaption,
 });
-/* END CARD ADD SECTION */
+popupImage.setEventListeners();
+/* END CARD VIEW IMAGE SECTION */
+
+/* CARD LIKE/UNLIKE SECTION */
+function handleCardLike(card) {
+  const request = card.getInfo().isLiked ? "unlikeCard" : "likeCard"; // inverted logic gets handled here
+  api[request](card.getInfo().id)
+    .then((cardRes) => {
+      card.likeCard(cardRes.isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+/* END CARD LIKE/UNLIKE SECTION */
+
+/* CARD DELETE SECTION */
+function handleCardPopupDelete(card) {
+  popupCardDelete.open(card);
+}
+const popupCardDelete = new PopupWithConfirm(
+  modalWindowCardDelete,
+  (event, card) => {
+    // this subclass acts like the form popup subclass except it has no inputs. It does however help to pass the necessary param to handle card deletion
+    event.preventDefault();
+    // console.log(card._cardElement);
+    api
+      .deleteCard(card.getInfo().id)
+      .then(() => {
+        card.deleteCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        popupCardDelete.close();
```

Comment:
* You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) **Please, correct it everywhere in the code**

Reply:
Please, move it into `then`